server/src/main/java/io/crate/planner/optimizer/rule/MergeFilterAndCollect.java

```diff
@@ -59,12 +60,12 @@ public Pattern<Filter> pattern() {
     @Override
     public LogicalPlan apply(Filter filter,
                              Captures captures,
-                             TableStats tableStats,
+                             PlanStatsProvider planStatsProvider,
                              TransactionContext txnCtx,
                              NodeContext nodeCtx,
                              Function<LogicalPlan, LogicalPlan> resolvePlan) {
         Collect collect = captures.get(collectCapture);
-        Stats stats = tableStats.getStats(collect.relation().tableInfo().ident());
+        Stats stats = planStatsProvider.tableStats().getStats(collect.relation().tableInfo().ident());
```

Comment:
Does it make sense to integrate the `SelectivityFunctions` directly into `PlanStatsProvider`?

Reply:
`PlanStatsProvider` can use `SelectivityFunctions` but I wouldn't inline the logic.